gitlabform/processors/project/project_processor.py

```diff
@@ -1,17 +1,76 @@
-from cli_ui import debug as verbose
-
+import traceback
+from logging import debug
+from cli_ui import debug as verbose, warning
 from gitlabform.gitlab import GitLab
 from gitlabform.processors.abstract_processor import AbstractProcessor
+from gitlabform.gitlab import GitlabWrapper
+from gitlab import Gitlab, GitlabGetError, GitlabTransferProjectError
+from gitlab.v4.objects import Project
 
 
 class ProjectProcessor(AbstractProcessor):
     def __init__(self, gitlab: GitLab):
         super().__init__("project", gitlab)
 
     def _process_configuration(self, project_and_group: str, configuration: dict):
-        if configuration["project"].get("archive"):
-            verbose("Archiving project...")
-            self.gitlab.archive(project_and_group)
-        else:
-            verbose("Unarchiving project...")
-            self.gitlab.unarchive(project_and_group)
+        if configuration["project"].get("transfer_from") is not None:
+            source_project_path_with_namespace = configuration["project"].get(
+                "transfer_from"
+            )
+            destination_project_path_with_namespace = project_and_group
+            gl: Gitlab = GitlabWrapper(self.gitlab)._gitlab
+
+            # Check if the project was already transfered (i.e. in previous run) or a project with same path already exists
+            try:
+                project: Project = gl.projects.get(
+                    destination_project_path_with_namespace
+                )
+                verbose(
+                    "Project already exists: '"
+                    + project.path_with_namespace
+                    + "'. Ignoring 'transfer_from' config..."
+                )
+            except GitlabGetError:
+                # Project doesn't exist at the destination. Let's process the transfer request
+                project_to_be_transferred: Project = gl.projects.get(
+                    source_project_path_with_namespace
+                )
+                destination_project_path = project_and_group.split("/")[-1]
+                # Check if the project path needs to be updated; In Gitlab, path maybe different than name
+                if destination_project_path != project_to_be_transferred.path:
+                    verbose(
+                        "Updating the source project path from '"
+                        + project_to_be_transferred.path
+                        + "' to '"
+                        + destination_project_path
```

Comment:
Nit: single quotation mark is not closed here.

Reply:
You might consider replacing those multiple lines with f-strings, which might make the code more readable and such mistakes more rare.

Reply:
Thank you. I'll look into it.

Reply:
I've pushed an update to address it. Please have a look.